tool/tsh/tsh.go

```diff
@@ -605,14 +605,24 @@ func onLogin(cf *CLIConf) {
 			tc.Logout()
 			utils.FatalError(err)
 		}
-		for _, roleName := range roleNames {
-			role, err := tc.GetRole(cf.Context, roleName)
-			if err != nil {
-				tc.Logout()
-				utils.FatalError(err)
+		// load all roles with re-used root auth client and collect
+		err = tc.WithRootClusterClient(cf.Context, func(clt auth.ClientI) error {
```

Comment:
Not sure this function is required, why not just do something like:

```go
authClient, err := ConnectToRootCluster(...)
[...]
role, err := authClient.GetRole(...)
[...]
```

Reply:
Each method on `TeleportClient` closes all connections before returning.  Likely because `TeleportClient` is used in such a way that its internal config may be changed significantly between method calls.  `WithRootClusterClient` maintains that pattern, and also lets me get around writing a custom interface for `auth.ClientI + io.Closer`.

Reply:
Okay, makes sense.